### PR TITLE
Removing redundant go-source tag

### LINKED ISSUE #83: go-source tag is no longer necessary
The package template generates a `go-source` tag along with the `go-import` tag.

https://github.com/uber-go/sally/blob/434fc43c4b5403b38db408621a286096806c4036/templates/package.html#L4-L5

The `go-import` tag is used by the `go` CLI to find the source code. Details at https://go.dev/ref/mod#vcs-find.

However, `go-source` is a leftover from pre-Go-module days, in service of godoc.org: https://github.com/golang/gddo/wiki/Source-Code-Links
This is no longer necessary -- pkg.go.dev serves documentation and module-version-aware source links directly.

### PR DESCRIPTION
**Pull request by SoulPancake**
- Resolves #83

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
Comment (abhinav) [inline .idea/.gitignore:1]: Would you please remove the .idea directory from the PR?
Feel free to add `/.idea` to the root `.gitignore`.

File: templates/package.html [hunk 2]
```diff
@@ -2,7 +2,6 @@
 <html>
     <head>
         <meta name="go-import" content="{{ .CanonicalURL }} git https://{{ .Repo }}">
-        <meta name="go-source" content="{{ .CanonicalURL }} https://{{ .Repo }} https://{{ .Repo }}/tree/{{ .Branch }}{/dir} https://{{ .Repo }}/tree/{{ .Branch }}{/dir}/{file}#L{line}">
```
Comment (abhinav) [inline templates/package.html:5]: FYI, with this removed, we also don't need the Branch field.
We can drop the Branch field from the template data passed in at line 131 in handler.go.
(We can also drop the corresponding config field, but that can be deferred.)

**Review by abhinav — COMMENTED** [review]
Thanks for the PR, @SoulPancake.
Left a couple minor suggestions.

File: handler.go [hunk 3]
```diff
@@ -50,7 +50,7 @@ type indexHandler struct {
 
 type packageInfo struct {
 	Desc       string // package description
-	ImportPath string // canonical improt path
+	ImportPath string // canonical import path
```
Comment (abhinav) [inline handler.go:53]: oops! thanks for fixing.

**Review by abhinav — APPROVED** [review]

**Comment by abhinav** [comment]
Awesome, thanks! 

CC @sywhang

**Comment by abhinav** [comment]
@SoulPancake The test failure appears to be just a tabs vs spaces in the example files. 
When you updated the test cases, you may have accidentally changed the expected output.

**Review by sywhang — APPROVED** [review]

**Pull request merged**